docs/docs/developers/docs/concepts/accounts/index.md

```diff
@@ -2,170 +2,362 @@
+- **Pre-funding**: Send funds to an address before the account is deployed
+- **Counterfactual deployment**: Interact with an account as if it exists, deploy it later
+- **Address recovery**: Recompute addresses from known keys
 
-For example, a lottery contract, where at some point a prize needs to be paid out to its winners. This `pay` action does not require authentication and does not need to be executed by any user in particular, so anyone could submit a transaction that defines the lottery contract itself as `origin` and `pay` as `entrypoint` function. However, it's on the contract to define how fees for the prize claim will be paid as they won't be paid by the account contract.
+#### Complete Address
 
-For an example of this behavior see our [e2e_crowdfunding_and_claim test](https://github.com/AztecProtocol/aztec-packages/blob/88b5878dd4b95d691b855cd84153ba884adf25f8/yarn-project/end-to-end/src/e2e_crowdfunding_and_claim.test.ts#L322) and the [SignerLess wallet](https://github.com/AztecProtocol/aztec-packages/blob/master/yarn-project/aztec.js/src/wallet/signerless_wallet.ts) implementation. Notice that the Signerless wallet doesn't invoke an `entrypoint` function of an account contract but instead invokes the target contract function directly.
+While an address alone is sufficient for receiving funds, spending notes requires a **complete address** which includes:
+
+- All the user's public keys (nullifier, incoming viewing, etc.)
+- The partial address (contract deployment information)
+- The contract address itself
+
+The complete address proves that the nullifier key inside the address is correct, enabling the user to spend their notes.
+
+## Keys and Privacy
+
+Aztec accounts use multiple specialized key pairs instead of a single key like traditional blockchains. This separation enables powerful privacy features:
+
+- Different keys for different purposes (spending, viewing, authorization)
+- Per-application key isolation for damage limitation
+- Flexible permission models without compromising security
+
+For detailed information about the four key types (nullifier, incoming viewing, address, and signing keys) and how they work together, see the [Keys documentation](./keys.md).
+
+## The Entrypoint Pattern
+
+The entrypoint is the gateway to your account. When someone wants to execute a transaction from your account, they call the entrypoint with a payload describing what to do.
+
+### Transaction Flow
+
+Here's how a transaction flows through an account:
+
+```mermaid
+sequenceDiagram
+    participant User
+    participant Wallet
+    participant Account Contract
+    participant Target Contracts
+
+    User->>Wallet: Request transaction (transfer, swap, etc.)
+    Wallet->>Wallet: Create payload & sign
+    Wallet->>Account Contract: Call entrypoint(payload, signature)
+    Account Contract->>Account Contract: Verify signature
+    Account Contract->>Account Contract: Pay fees
```

Comment:
Would be good to confirm, but I think it would make sense to put the "pay fees" arrow after "execute calls", since fees aren't actually known until after all of the code is executed. The simplification may also be a little confusing since there are fees from private execution and public execution. 

<img width="1952" height="1344" alt="image" src="https://github.com/user-attachments/assets/6d65b007-725d-4660-ab7b-9ded7c82121f" />